### ISSUE TITLE
Feedback and current project status

### ISSUE THREAD

**Issue by mrddter** [andrewbaronick/BacktestJS#2]
Hello.

First of all, congratulations! This project is really well written and is exactly what is needed to thoroughly test a trading strategy.

Below are some notes, which I hope will be read in a positive light as they are intended to be:

- Useful: specify that the compatible version of Node is 18 (it seems not to work with version 22)
- Useful: an explanation or a video explanation of the various columns and fields (of the result on the screen/browser)
- New request: implement trailingStopLoss
- New request: ability to define how many orders to have in parallel (e.g., a maximum of 3, etc.)
- New request: use the NPM method to create projects with custom strategies (ts). For example, having the npm core library separate from the npm global to run it standalone
- Low priority: indicate profit or loss in "all orders" with colors or symbols (e.g., colored arrows or others)
- Low priority: an example that allows creating orders with takeProfit would be useful

That said, congratulations again. I hope the project continues to progress.

**Comment by mrddter** [comment]
For now, I have forked the project to keep it maintained. I hope it is appreciated.
You can find it [here](https://github.com/backtestjs/backtestjs).

**Comment by andrewbaronick** [comment]
@mrddter  Thanks for the notes and happy to hear your interested in the project!  

The notes are great and most i agree with, i might ask that you keep the code inside this project and put PR's into here if wanting to update but of course if you want to maintain on your own fork i support it as well!

As you can see im not the quickest to update but i will definitely update at a minimum once a month to questions / issues / pr's, just been no one really interested since i put the project out to maintain so much.

In general ive been thinking of a few ways to market the project a little more which i think might motivate me more to code more features ive really wanted to add which will hopefully lead to better maintenance haha.

**Comment by mrddter** [comment]
Hello Andrew, sorry I missed your reply. 

I've been making quite a few changes based on your work. Essentially, I've split the project into two parts so I can use it as a core library. The CLI interface part is still in a separate repo (it's great for speed and the reporting feature, which is definitely valuable).

For clarity and simplicity, I've created an organization (https://github.com/orgs/backtestjs/repositories) with 3 separate repos to keep everything well-organized, allow for evolution, and stick to your principles (e.g., open source, freely usable, etc.):

- The core framework with all the logic
- The very useful CLI part
- A quick-start guide for anyone who wants to use the library in a project (which is actually how I'm using it). This repo is also a GitHub template, making it easy to fork or replicate if anyone needs it.

At the moment, I don't plan to develop it much further because I've already added some features that I consider sufficient for my needs:

- Scanning strategies without worrying about adding, deleting, or updating existing ones
- Running strategies on a specific symbol or interval while using historical support data as a "stream". For example, I trade on a 1-hour interval but also evaluate what happens statistically on different time frames (30m, 1d, etc.)
- In `runStrategy`, all this information is now available, providing greater granularity (which was necessary for my purposes)
- I replaced `tulind` with `technicalindicators`, which I know quite well, and it allows me to use Node v22, for example
- Added some colors and small adjustments to the generated HTML file, including a notes column on orders to specify useful indicators during the review phase
- Revised file structures and various logics (so I expect some bugs for sure)
- And more, but I can't remember everything (I should have documented everything in the various readmes anyway)
- I haven't implemented trailingStopLoss yet, but I will because I need it

Personally, I think it's a bit early to consider monetizing the project, although I'm definitely interested. For now, as I mentioned, I'm using it as a foundation for my work, but if you'd like, we can discuss it together in pvt. Let me know.

Thanks again!
